Lib/profiling/sampling/gecko_collector.py

```diff
@@ -0,0 +1,451 @@
+import json
+import os
+import platform
+import sys
+import time
+from typing import Dict, List, Optional, Tuple, Any
+
+from .collector import Collector, THREAD_STATE_RUNNING
+
+
+class GeckoCollector(Collector):
+    def __init__(self, *, skip_idle=False):
+        self.skip_idle = skip_idle
+        self.start_time = time.time() * 1000  # milliseconds since epoch
+
+        # Global string table (shared across all threads)
+        self.global_strings = ["(root)"]  # Start with root
+        self.global_string_map = {"(root)": 0}
+
+        # Per-thread data structures
+        self.threads = {}  # tid -> thread data
+
+        # Global tables
+        self.libs = []
+
+        # Sampling interval tracking
+        self.sample_count = 0
+        self.last_sample_time = 0
+        self.interval = 1.0  # Will be calculated from actual sampling
```

Comment:
ℹ️ **Maintainability**

The initial value of `self.interval` is set to 1.0, but it is not clear why this value is chosen. Consider adding a comment explaining the choice of this initial value or calculate it based on a more meaningful metric.